src/Azure.Functions.Cli/Actions/KubernetesActions/KubernetesDeployAction.cs

```diff
@@ -102,21 +109,27 @@ public override async Task RunAsync()
                 triggers = await DockerHelpers.GetTriggersFromDockerImage(resolvedImageName);
             }
 
-            var resources = KubernetesHelper.GetFunctionsDeploymentResources(
+            var functionNames = triggers.FunctionsJson.Select(f => f.Key);
+            var funcAppKeys = FuncAppKeysHelper.CreateKeys(functionNames);
```

Comment:
I think this can be encapsulated in the `GetFunctionsDeploymentResources` call to keep the action itself simpler.

Reply:
also keys are only needed for http functions. In `GetFunctionsDeploymentResources`, functions are split into 2 collections, and we only need to create the keys for http ones